src/metabase/models/setting.clj

```diff
@@ -530,6 +532,13 @@
                       (core/get cache (setting-name setting-definition-or-name))))))]
         (not-empty v)))))
 
+(defn realize
+  "Parsing a setting may result in a lazy value. Use this to ensure we finish parsing."
```

Comment:
This surprised me. When can parsing a setting result in a lazy value?

Reply:
Yeah, it surprised me too. 

A json list, which I would expect to be parsed as a vector (matching JS semantics), returns not only a lazy list, that list is also parsed lazily. 

This seems absurd, given that you could close the stream that was parsed.

Reply:
Just to double check my sanity, I took the call to this function out again and confirmed that we stop throwing the exception in two of the test cases. 🤯

Reply:
Nice find. It feels like that should be fixed inside the implementation of the `parse-fn`. `realize` shouldn't have to deal with this. You could update the default implementation of `get-value-of-type` `:json` to use the `json/parse-string-strict` to avoid lazily parsing

Reply:
On the other hand, I'm a bit torn. Someone could easily write a getter for a setting that accidentally parses lazily without realising. maybe we should leave this.

Reply:
Note to self: read doc strings more often 😆

Reply:
Yes!